### ISSUE TITLE
Roadmap for NodeJS updates?

### ISSUE THREAD

**Issue by fkusei** [bitfocus/companion#2333]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Companion currently requires NodeJS `>=14.18 <15`, which according to the [endoflife.date information about NodeJS](https://endoflife.date/nodejs) is out of Active Support since 2021-10-19 (but getting security support until 2023-04-30). Even NodeJS 16 is out of active Support since 2022-10-18.

Are you planning to update Companion to be compatible to more recent NodeJS versions (preferrably 18+)?

**Comment by Julusian** [comment]
The 3.0 betas are using 18, and going forward it will be easier for us to update the version gradually across the modules.
In 2.x we were also limited by electron, which is also no longer the case

**Comment by fkusei** [comment]
Thanks for the quick response!

Is there ETA for getting 3.0 out? I see it is currently being worked on in https://github.com/orgs/bitfocus/projects/2 but cannot find any information about a planned release (or even a beta).
I have been approached by the security people at work, because we're still running NodeJS 14 for companion, which is why i'm asking about this.

**Comment by Julusian** [comment]
There isnt currently an ETA, but it is something I am thinking needs to be chased.
We are currently a little blocked on 10% of the modules being broken in 3.0 https://github.com/bitfocus/companion/issues/2157

**Comment by DanielBaulig** [comment]
@Julusian: You know what (typically) does wonders to encourage module / dependent developers migrate their code to non-backwards compatible updates? Giving them a deadline for when you will break their sh*t. :)

**Comment by Julusian** [comment]
@DanielBaulig yeah, but its going to be the users who suffer and its a bad look for the project and a release that will bring a lot more stability if too many modules dont work.

Nothing has been agreed for certain, but it has been suggested to hopefully have an RC out in mid april.
